pkg/storage/allocator_scorer.go

```diff
@@ -238,9 +249,9 @@ var _ sort.Interface = byScoreAndID(nil)
 
 func (c byScoreAndID) Len() int { return len(c) }
 func (c byScoreAndID) Less(i, j int) bool {
-	if c[i].diversityScore == c[j].diversityScore &&
+	if scoresAlmostEqual(c[i].diversityScore, c[j].diversityScore) &&
```

Comment:
An inconsistent `Less` function could cause all kinds of chaos. Is there any risk of floating-point math causing problems here? I think it might be better to sort things according to the raw float values, then apply the filter at the end so we won't swap to a node that is not a sufficient improvement.